dev/NavigationView/readme.md

```diff
@@ -0,0 +1,51 @@
+# NavigationView Control
+
+NavigationView is a powerful control that makes it easy to integrate a rich navigation experience into apps.
+This document outlines overall structure and behaviors.
+Information related to the NavigationViewItem, or the rendering of the control see the documents in the /NavigationView/docs folder.
+
+## Architecture
+
+NavigationView is a complex control, which has a lot of internal states to store.
+
+Rendering of the menu items is being done through three ItemsRepeaters, one for the NavigationView in left mode, one for the NavigationView in top mode and one for the overflow menu.
```

Comment:
I think a better way to outline this (to include hierarchy and now footer items) is to make a distinction between the root repeaters we have (the ones defined in `NavigationView::IsRootItemsRepeater`) and then mention the auto generated repeaters in each NVI (which are used for hierarchy)

Reply:
Updated that section now.